cumulusci/tasks/marketing_cloud/deploy.py

```diff
@@ -0,0 +1,105 @@
+}
+
+
+class MarketingCloudDeployTask(BaseMarketingCloudTask):
+
+    task_options = {
+        "package_zip_file": {
+            "description": "The unzipped contents of the package to deploy.",
+            "required": True,
+        }
+    }
+
+    def _run_task(self):
+        pkg_zip_file = Path(self.options["package_zip_file"])
+        if not pkg_zip_file.is_file():
+            self.logger.error(f"Package zip file not valid: {pkg_zip_file.name}")
+            return
+
+        with temporary_dir() as temp_dir:
+            with zipfile.ZipFile(pkg_zip_file) as zf:
+                zf.extractall(temp_dir)
+                payload = self._construct_payload(
+                    Path(f"{temp_dir}/{pkg_zip_file.stem}")
+                )
+
+        response = requests.post(
+            MC_DEPLOY_ENDPOINT,
+            json=payload,
+            headers={
+                "Authorization": f"Bearer {self.mc_config.access_token}",
+                "SFMC-TSSD": self.mc_config.tssd,
+            },
+        )
+
+        self._validate_response(response)
+
+    def _construct_payload(self, dir_path):
+        dir_path = Path(dir_path)
+        assert dir_path.is_dir(), "package_directory must be a directory"
+
+        payload = defaultdict(lambda: defaultdict(dict))
+        payload["namespace"] = PAYLOAD_NAMESPACE_VALUES
+        payload["config"] = PAYLOAD_CONFIG_VALUES
+
+        with open(dir_path / "references.json", "r") as f:
+            payload["references"] = json.load(f)
+
+        with open(dir_path / "input.json", "r") as f:
+            payload["input"] = json.load(f)
+
+        entities_dir = Path(f"{dir_path}/entities")
+        for item in entities_dir.glob("**/*.json"):
+            if item.is_file():
+                entity_name = item.parent.name
+                entity_id = item.stem
+                with open(item, "r") as f:
+                    payload["entities"][entity_name][entity_id] = json.load(f)
+
+        return payload
+
+    def _validate_response(self, response: requests.Response):
+        """Checks for any errors present in the response to the deploy request.
+        Displays errors if present, else informs use that the deployment was successful."""
+
+        if response.status_code != 200:
+            raise CumulusCIException(response.text)
+
+        has_error = False
+        deploy_info = json.loads(response.text)
+        for entity, info in deploy_info["entities"].items():
+            if not info:
+                continue
+
+            for entity_id, info in info.items():
+                if info["status"] == "FAILED":
+                    has_error = True
+                    self.logger.error(
+                        f"Failed to deploy {entity}/{entity_id}: {info['issues']}"
+                    )
+                elif info["status"] == "SKIPPED":
+                    has_error = True
+                    self.logger.warn(f"Skipped deploying entity: {entity}/{entity_id}")
```

Comment:
It would be safer to check for any status that is not SUCCESS, so that we don't report success if a new status is added that we don't know about.